src/py_obs/osc.py

```diff
@@ -15,8 +17,10 @@ def __str__(self) -> str:
 class Osc:
     username: str
     password: str
+    private_key_path: str
```

Comment:
Now that we support either password based auth **or** private key based auth, we don't need both parameters at the same time. However, we should check in `__post_init__` that one **and only** one of them is set.
```suggestion
    password: str | None = None
    private_key_path: str | None = None
```